naming/src/main/java/org/jboss/as/naming/WildFlyRootContextWrapper.java

```diff
@@ -0,0 +1,346 @@
+    }
+
+    @Override
+    public NamingEnumeration<NameClassPair> list(String nameString) throws NamingException {
+        return delegate.list(nameString);
+    }
+
+    @Override
+    public NamingEnumeration<Binding> listBindings(Name name) throws NamingException {
+        return delegate.listBindings(name);
+    }
+
+    @Override
+    public NamingEnumeration<Binding> listBindings(String nameString) throws NamingException {
+        return delegate.listBindings(nameString);
+    }
+
+    @Override
+    public void destroySubcontext(Name name) throws NamingException {
+        delegate.destroySubcontext(name);
+    }
+
+    @Override
+    public void destroySubcontext(String nameString) throws NamingException {
+        delegate.destroySubcontext(nameString);
+    }
+
+    @Override
+    public Context createSubcontext(Name name) throws NamingException {
+        return delegate.createSubcontext(name);
+    }
+
+    @Override
+    public Context createSubcontext(String nameString) throws NamingException {
+        return delegate.createSubcontext(nameString);
+    }
+
+    @Override
+    public Object lookupLink(Name name) throws NamingException {
+        aroundLookup(name);
+        return delegate.lookupLink(name);
+    }
+
+    @Override
+    public Object lookupLink(String nameString) throws NamingException {
+        Name name = NAME_PARSER.parse(nameString);
+        return lookupLink(name);
+    }
+
+    @Override
+    public NameParser getNameParser(Name name) throws NamingException {
+        return delegate.getNameParser(name);
+    }
+
+    @Override
+    public NameParser getNameParser(String nameString) throws NamingException {
+        return delegate.getNameParser(nameString);
+    }
+
+    @Override
+    public Name composeName(Name name, Name name1) throws NamingException {
+        return delegate.composeName(name, name1);
+    }
+
+    @Override
+    public String composeName(String nameString, String name1) throws NamingException {
+        return delegate.composeName(nameString, name1);
+    }
+
+    @Override
+    public Object addToEnvironment(String nameString, Object object) throws NamingException {
+        return delegate.addToEnvironment(nameString, object);
+    }
+
+    @Override
+    public Object removeFromEnvironment(String nameString) throws NamingException {
+        return delegate.removeFromEnvironment(nameString);
+    }
+
+    @Override
+    public Hashtable<?, ?> getEnvironment() throws NamingException {
+        return delegate.getEnvironment();
+    }
+
+    @Override
+    public void close() throws NamingException {
+        delegate.close();
+    }
+
+    @Override
+    public String getNameInNamespace() throws NamingException {
+        return delegate.getNameInNamespace();
+    }
+
+    private final void aroundLookup(Name name) throws NamingException {
+        for (LookupInterceptor interceptor : interceptors) {
+            interceptor.aroundLookup(name);
+        }
```

Comment:
This is prone to ConcurrentModificationExceptions.
Since the list implementation is a synchronized collection wrapper, iteration is only thread-safe within a synchronized block.
If updates are sufficiently infrequent (and if I understand this PR correct, this seems to be the case), I would instead opt for a COW-based concurrent data structure, e.g. CopyOnWriteArrayList.

Reply:
Agree, this is a bug. The list itself is mean to be modified during the server startup in most (if not all scenarios), so CopyOnWriteArrayList seems like a good solution. I have updated the PR.